packages/main/src/DateTimePicker.ts

```diff
@@ -210,9 +203,8 @@ class DateTimePicker extends DatePicker {
 	 * @public
 	 */
 	async openPicker(): Promise<void> {
-		await super.openPicker();
-		this._currentTimeSlider = "hours";
 		this._previewValues.timeSelectionValue = this.value || this.getFormat().format(new Date());
```

Comment:
also please check `_calendarSelectedDates` and `_calendarTimestamp`. Seems that the fallback is always executed and the overrides seems to be redundant.

Reply:
After you found the root of the problem, "reordering" is fixed.

the mentioned getters _calendarSelectedDates and _calendarTimestamp are necessary, you can check it by commenting them and try to use a picker when there is no date selected already.

Reply:
this._previewValues. _calendarSelectedDates is never set so we always use the value which comes from the parent class. Take this as possible cleanup in separate PR

Reply:
It is set in the onSelectedDatesChange method